rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,226 @@
+We want to have a single version for the "public" Kubewarden
+components and use
+[Semantic Versioning](https://semver.org/) as a way to express it.
+
+The controller, Policy Server and kwctl components are going to share the same Major and Minor version numbers. Each project is
+going to have an independent patch level number.
+
+All the non-visible components of Kubewarden (e.g. the different
+Rust libraries) and the policies are going to retain their
+existing version numbers, release strategy and independence.
+
+### Helm charts
+
+Helm charts have two kind of version numbers:
+
+* `version`: a SemVer 2 version specific to the helm chart
+* `appVersion`: the version of the app that the chart contains
+
+The helm charts will keep their own independence when it comes to
+the `version` attribute, while the `appVersion` attribute
+will always be set to the `<Major>.<Minor>` version of the
+Kubewarden stack.
+
+> See the official documentation about
+[`Chart.yaml`](https://helm.sh/docs/topics/charts/#the-chartyaml-file)
+for more information.
+
+## Examples
+
+This section provides clarity about how the proposal would work
+using some concrete real-world scenarios.
+
+### A new release Kubewarden takes place
+
+A new version of Kubewarden stack has to be released. That can
+happen because a new important feature/bugfix has been added,
+because of marketing reasons,...
+
+Let's assume:
+
+* The current version of the Kubewarden stack is `0.5.3`
+* The current version of the helm chart is `v0.10.3`
+* The new feature is not worth a major bump
+
+Hence we will perform a minor update.
+
+We are going to tag `v0.6.0` of all these components and release them:
+
+* kubewarden controller
+* policy server
+* kwctl
+
+This will happen **regardless** of the changes done to these projects. For example, kwctl is going to have a new release `v0.6.0` even if it didn't have any new commit since `v0.5.3`.
+
+The helm chart is updated in this way:
+
+* The version of container images of policy server and kubewarden
+  controller are upgraded to be `v0.6.0`
+* It's `version` attribute is set to `v0.10.4` because no significant changed happened
+* The `appVersion` attribute is set to `0.6.0`
+
+### A patch is done to kwctl
+
+A series of minor issues are found inside of kwctl. We decide to
+tag a new release that includes these fixes.
+
+The version of `kwctl` is updated from `0.6.0` to `0.6.1`.
+No other releases are done
+
+### A patch is done to kubewarden controller
+
+A minor issue is fixed with kubewarden controller. We decide to
+tag a new release that includes the fix.
+
+The following actions are done:
+
+* We tag a new release of kubewarden-controller: `0.6.1`
+* We update the helm chart that makes use of this image:
+  * We update the references of the image to ensure `0.6.1` is being pulled
+  * We update the `version` number of the chart itself to be
+    change from `0.10.4` to `0.10.5`.
+    
+    **Note:** we don't change the `appVersion` attribute
+
```

Comment:
Okay, if I understand correctly we could end with multiple users using the same `appVersion` (e.g. `0.6.0`) but actually running different container images. Right?

Reply:
yes with different patch level images. We need this if we want each project to have an independent patch level number

Reply:
How would users then report the version (of each component) in a bug report ? Reporting the chart version or the appVersion is not sufficient.
Note that users might only know the GUI, without ever having seen `kwctl`.

Reply:
If the user tells us the `version` of the helm chart he has installed, we will be able to determine the exact versions of the container images being deployed.

Reply:
The impression that I get is, in the end of the day, the only version that it will be used  is the helm chart version. Because this is the version important to solve the issues. The `appVersion` will be used only for marketing and announcements.